src/coralogix.js

```diff
@@ -147,13 +147,21 @@ class CoralogixLogger extends FormattedLoggerBase {
         subsystemName: subsystem,
         computerName: host,
         logEntries: [{
-          timestamp: Number(BigDate.preciseTime(fields.timestamp)),
+          timestamp: CoralogixLogger._preciseTime(fields.timestamp),
           text: payload,
           severity: _logLevelMapping[fields.level],
         }],
       },
     });
   }
+
+  static _preciseTime(ts) {
+    try {
+      return Number(BigDate.preciseTime(ts));
+    } catch (e) {
+      return new Date().getTime();
```

Comment:
can we log the error here? so we can find out what causes this error.

Reply:
where do you suggest the error to be logged? we're already inside a calling sequence that tries to log an error via HTTP.

Reply:
We could just wrap BigDate::preciseTime into a try/catch error and rethrow?

```
try {
  …
catch (e) {
  const outer_e = new BigDateError(`Failed to parse bigDate from value ${val}: ${e.message}`);
  assign(outer_e, { val, cause: e });
  throw outer_e;
}
```

Reply:
In my experience, this either hides the actual log message or makes it log to an unexpected location (e.g. `infrastructure`).

Instead, I provided a patch to append the offending timestamp to the message being logged, which should help the client of `helix-log` to locate the code passing that timestamp.

Reply:
Please ignore that patch: payload can be either a string or JSON, so it wouldn't work in the latter case.

Reply:
Rethrowing in the case of `CoralogixLogger` will not work, as it catches errors thrown by `_sendRequest` and tries again.